panther_driver/src/panther_hardware.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/python3
+
+import rospy
+from std_srvs.srv import SetBool, SetBoolRequest, SetBoolResponse
+from std_srvs.srv import Trigger, TriggerRequest, TriggerResponse
+
+import os
+import time
+import threading
+# import multiprocessing
+
+import RPi.GPIO as GPIO
+from gpiozero import Button, PWMOutputDevice, OutputDevice
+from signal import pause
+from std_msgs.msg import Bool
+
+# Define pin names
+VMOT_ON = 6
+CHRG_SENSE = 7
+WATCHDOG = 14
+FAN_SW = 15
+# SHDN_INIT = 16 Shutdown Init managed by systemd service
+AUX_PW_EN = 18
+CHRG_EN = 19
+VDIG_OFF = 21
+DRIVER_EN = 23
+E_STOP_RESET = 27
+
+class Watchdog:
+    def __init__(self) -> None:
+        self.watchdog_on = False
+        self.watchdog_pwm = PWMOutputDevice(WATCHDOG)
+
+    def turn_on(self):
+        if not self.watchdog_on:
+            freqency = 50
+            step = 1/freqency/2
+            self.watchdog_pwm.blink(on_time=step,off_time=step)
+            self.watchdog_on = True
+
+    def turn_off(self):
+        if self.watchdog_on:
+            self.watchdog_pwm.off()
+            self.watchdog_on = False
+
+
+class PantherHardware:
+    def __init__(self) -> None:
+        # Setup GPIO pins
+        GPIO.setmode(GPIO.BCM)
+        GPIO.setup(VMOT_ON, GPIO.OUT, initial=0)
+        GPIO.setup(CHRG_SENSE, GPIO.IN)
+        GPIO.setup(FAN_SW, GPIO.OUT, initial=0)
+        GPIO.setup(AUX_PW_EN, GPIO.OUT, initial=0)
+        GPIO.setup(CHRG_EN, GPIO.OUT, initial=1)
+        GPIO.setup(VDIG_OFF, GPIO.OUT, initial=0)
+        GPIO.setup(DRIVER_EN, GPIO.OUT, initial=0)
+        GPIO.setup(E_STOP_RESET, GPIO.IN) # USED AS I/O
+
+        self.watchdog = Watchdog()
+        self.watchdog.turn_on()
+
+        # Setup ROS Node
+        rospy.init_node('panther_hardware')
+
+        # ROS Services
+        self.aux_power_enable_service = rospy.Service('/panther_hardware/aux_power_enable', SetBool, self.handle_aux_power_enable)
+        self.charger_enable_service = rospy.Service('/panther_hardware/charger_enable', SetBool, self.handle_charger_enable)
+        self.disable_digital_service = rospy.Service('/panther_hardware/disable_digital_power', SetBool, self.handle_disable_digital_power)
+        self.motors_enable_service = rospy.Service('/panther_hardware/motors_enable', SetBool, self.handle_motors_enable)
+        self.fan_enable_service = rospy.Service('/panther_hardware/fan_enable', SetBool, self.handle_fan_enable)
+        self.reset_e_stop_service = rospy.Service('/panther_hardware/reset_e_stop', Trigger, self.handle_reset_e_stop)
+        self.trigger_e_stop_service = rospy.Service('/panther_hardware/trigger_e_stop', Trigger, self.handle_trigger_e_stop)
```

Comment:
The SetBool services are ment to be as follows. EG. motors_enable is used to enable motors using true to enable, false to disable. As simple as this.
For e-stop we need to have to trigger services because reset and trigger is not as simple as toggling one pin as with other services